artifactory/commands/golang/go.go

```diff
@@ -330,19 +330,25 @@ func buildPackageVersionRequest(name, branchName string) string {
 	return path.Join(packageVersionRequest, "latest.info")
 }
 
-func SetArtifactoryAsResolutionServer(serverDetails *config.ServerDetails, depsRepo string) (err error) {
-	err = setGoProxy(serverDetails, depsRepo)
+func SetArtifactoryAsResolutionServer(serverDetails *config.ServerDetails, depsRepo string, isCurationCmd bool) (err error) {
+	err = setGoProxy(serverDetails, depsRepo, isCurationCmd)
 	if err != nil {
 		err = fmt.Errorf("failed while setting Artifactory as a dependencies resolution registry: %s", err.Error())
 	}
 	return
 }
 
-func setGoProxy(server *config.ServerDetails, remoteGoRepo string) error {
-	repoUrl, err := goutils.GetArtifactoryRemoteRepoUrl(server, remoteGoRepo)
+func setGoProxy(server *config.ServerDetails, remoteGoRepo string, isCurationCmd bool) error {
+	repoUrl, err := goutils.GetArtifactoryRemoteRepoUrl(server, remoteGoRepo, isCurationCmd)
 	if err != nil {
 		return err
 	}
-	repoUrl += "|direct"
+	if !isCurationCmd {
+		repoUrl += "|direct"
+	}
 	return os.Setenv("GOPROXY", repoUrl)
 }
+
+func SetGoModCache(cacheFolder string) error {
+	return os.Setenv("GOMODCACHE", cacheFolder)
+}
```

Comment:
```suggestion
const GoModCacheEnvVar = "GOMODCACHE"
func SectGoModCache(cacheFolder string) (err error, restore func() error) {
	oldVal, exists := os.LookupEnv(GoModCacheEnvVar)
	return errorutils.CheckError(os.Setenv(GoModCacheEnvVar, cacheFolder)), func() error {
		if !exists {
			return errorutils.CheckError(os.Unsetenv(GoModCacheEnvVar))
		} else {
			return errorutils.CheckError(os.Setenv(GoModCacheEnvVar, oldVal))
		}
	}
}
```
You can add option to restore env when done, most of the times when we end our commands we want to restore the user env

Reply:
it shouldnt change the user env, only in this session maybe, but I'm not sure